### PR TITLE
Remove responses

### PR DESCRIPTION
**Pull request by MrIvanPlays**
Addresses #256 and #258

Needed merge before #266 to make work on it easier

### CODE REVIEW

**Comment by MrIvanPlays** [comment]
and push didn't compile because of minestom...

File: api/src/main/java/me/lokka30/treasury/api/economy/currency/Currency.java [hunk 1]
```diff
@@ -167,15 +166,15 @@ default BigDecimal to(@NotNull Currency currency, @NotNull BigDecimal amount) {
      * If a per-locale formatting system is not used by the currency, the parameter can be ignored.
      * If the specified locale is null, the currency implementation should assume a default.
      * <p>
-     * If the specified string cannot be parsed, a {@link Response} with a suitable
-     * {@link me.lokka30.treasury.api.common.response.FailureReason} is returned.
+     * If the specified string cannot be parsed, a failure {@link CompletableFuture} shall be
+     * returned with a suitable message in the form of
```
Comment (lokka30) [inline api/src/main/java/me/lokka30/treasury/api/economy/currency/Currency.java:170]:
```suggestion
     * returned with a suitable message in the form of a
```

File: api/src/main/java/me/lokka30/treasury/api/economy/account/accessor/PlayerAccountAccessor.java [hunk 2]
```diff
@@ -56,21 +55,15 @@ public CompletableFuture<Response<PlayerAccount>> get() {
      * Does the same as what {@link #get()} does, but returns a generic {@link Account}
      * interface, rather a specific {@link PlayerAccount} interface.
      *
-     * @return future with response which if successful returns the resulting
-     *         {@link PlayerAccount} in the form of {@link Account}
+     * @return a player account in the form of {@link Account}
```
Comment (lokka30) [inline api/src/main/java/me/lokka30/treasury/api/economy/account/accessor/PlayerAccountAccessor.java:58]:
```suggestion
     * @return a player account in the form of an {@link Account}
```

File: api/src/main/java/me/lokka30/treasury/api/economy/account/accessor/NonPlayerAccountAccessor.java [hunk 3]
```diff
@@ -58,11 +58,10 @@ public NonPlayerAccountAccessor withName(@Nullable String name) {
     /**
      * Gets or creates the {@link NonPlayerAccount non player account} needed.
      *
-     * @return future with {@link Response} which if successful returns the resulting
-     *         {@link NonPlayerAccount}
+     * @return a resulting non player account
```
Comment (lokka30) [inline api/src/main/java/me/lokka30/treasury/api/economy/account/accessor/NonPlayerAccountAccessor.java:61]:
```suggestion
     * @return a resulting non-player account
```

File: platform/bukkit/src/main/java/me/lokka30/treasury/plugin/bukkit/hooks/papi/economy/BalanceCache.java [hunk 4]
```diff
@@ -104,52 +102,47 @@ private void proceed(
...
-                if (!res.getFailureReason().getDescription().equalsIgnoreCase("accountNotExists")) {
-                    // log the problem and proceed with next entry
-                    TreasuryPlugin.getInstance().logger().error(
-                            "Error whilst trying to update balance cache for " + (player.getName() != null
-                                    ? player.getName()
-                                    : player.getUniqueId().toString()) + ": " + res
-                                    .getFailureReason()
-                                    .getDescription());
+                if (ex instanceof TreasuryException) {
+                    if (!ex.getMessage().equalsIgnoreCase("accountNotExists")) {
```
Comment (lokka30) [inline platform/bukkit/src/main/java/me/lokka30/treasury/plugin/bukkit/hooks/papi/economy/BalanceCache.java:112]: `"accountNotExists"` should be a constant somewhere.

**Review by lokka30 — CHANGES_REQUESTED** [review]
A fantastic upgrade to Treasury. Thank you Ivan!

File: api/src/main/java/me/lokka30/treasury/api/economy/account/Account.java [hunk 5]
```diff
@@ -444,12 +425,15 @@ CompletableFuture<Response<TriState>> setPermissions(
      *
      * @param player      the {@link UUID} of the player to check if they have the permission
      * @param permissions the permissions to check
-     * @return future with {@link Response} which if successful returns whether the player has all the specified permissions
+     * @return a {@link TriState} value, indicating the permission(s) value(s) for the specified
+     *         permission(s). {@link TriState#TRUE} represents that the specified member has (all) the
+     *         specified permission(s), {@link TriState#FALSE} explicitly says that the specified member
+     *         does not have (all) the specified permission(s), whilst {@link TriState#UNSPECIFIED}
+     *         represents that there is/are no value(s) held for the specified permission(s).
```
[the remark this answers is not in the document]
Comment (MrIvanPlays) [reply api/src/main/java/me/lokka30/treasury/api/economy/account/Account.java:432]: yes UNSPECIFIED shall be returned

File: platform/bukkit/src/main/java/me/lokka30/treasury/plugin/bukkit/hooks/papi/economy/BalanceCache.java (hunk 4 above)
Comment (MrIvanPlays) [reply platform/bukkit/src/main/java/me/lokka30/treasury/plugin/bukkit/hooks/papi/economy/BalanceCache.java:112]: see above; it's used in `.thenCompose`. also this is not API

File: api/src/main/java/me/lokka30/treasury/api/economy/account/Account.java (hunk 5 above)
[the remark this answers is not in the document]
Comment (lokka30) [reply api/src/main/java/me/lokka30/treasury/api/economy/account/Account.java:432]: ok, let's specify that particular case in the javadoc 👍🏻

**Comment by MrIvanPlays** [comment]
@Jikoo can you do a review?

**Review by lokka30 — APPROVED** [review]
Fantastic!

**Pull request merged**